File: app/src/main/java/com/hzjytech/hades/desginpattern/MainActivity.java
package com.hzjytech.hades.desginpattern;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        Log.d("----","-----"+DeviceIdManager.getID(MainActivity.this,1)+"----"+DeviceIdManager.getID(MainActivity.this,2)+"------"+DeviceIdManager.getID(MainActivity.this,3));

        Toast.makeText(MainActivity.this,getString(R.string.app_name),Toast.LENGTH_LONG).show();
    }
}

File: kedditbystep/src/main/java/com/hzjytech/hades/kedditbystep/MainActivity.kt
package com.hzjytech.hades.kedditbystep

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import com.hzjytech.hades.kedditbystep.fatures.news.NewsFragment

/**
 * Created by Hades on 2017/12/2.
 */
class MainActivity: AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        changeFragment(NewsFragment())
    }

    private fun changeFragment(fragment: Fragment) {
        val ft=supportFragmentManager.beginTransaction();
        ft.replace(R.id.fl_container,fragment);
        ft.commit();
    }

}